src/Files.App/Views/Layouts/BaseLayoutPage.cs

```diff
@@ -423,6 +423,9 @@ protected override async void OnNavigatedTo(NavigationEventArgs e)
 				ParentShellPageInstance.InstanceViewModel.IsPageTypeSearchResults = false;
 				ParentShellPageInstance.ToolbarViewModel.PathControlDisplayText = navigationArguments.NavPathParam;
 
+				if (ZipStorageFolder.IsZipPath(workingDir))
+					ParentShellPageInstance.ToolbarViewModel.CanExtract = true;
+
```

Comment:
The original condition is not simply a check to see if it is inside a zip folder. This would prevent the Extract button from appearing when a zip file is selected.
`IsArchiveOpened ? (SelectedItems is null || !SelectedItems.Any()) : IsSelectionArchivesOnly`

Reply:
Oh understood but is this approach okay?